IceCreamShop.playground/Contents.swift

```diff
@@ -1 +1,91 @@
 
+enum Size: Double {
+    case small = 2.40
+    case medium = 3.50
+    case large = 5.00
+}
+
+struct Flavor {
+    var name: String
+    var rating: Float
+    
+    init(name: String, rating: Float) {
+        self.name = name
+        self.rating = rating
+    }
+}
+
+struct Cone {
+    var flavor: String
+    var topping: String?
+    var size: Size
+    
+
+    func eat() {
+        print("chocolate")
+    }
+}
+
+
+class IceCreamShop {
+    var toppings: [String]
+    var sizes: [Size]
+    var flavors: [Flavor]
+    var totalSales: Double = Double()
+    
+    
+    init() {
+        self.toppings = ["Caramel", "Green tea", "Rocky road"]
+        self.sizes = [.small, .medium, .large]
+        self.flavors = [Flavor(name: "Chocolate", rating: 4.0), Flavor(name: "Pinacolada", rating: 5.0), Flavor(name: "Butter Pecan", rating: 6.0), Flavor(name: "Vanilla", rating: 2.0), Flavor(name: "Strawberry", rating: 3.0)]
+    }
+    
+        
+    
+    func listTopFlavors() {
+       var topFlavorNames = [String]()
+        
+        for flavor in self.flavors {
+            if flavor.rating > 4.0 {
+                topFlavorNames.append(flavor.name)
+            }
+        }
+        print(topFlavorNames)
+    }
+    
+    func orderCone(cone: Cone) -> Cone? {
+        let cone = Cone(flavor: "Vanilla", topping: "Yellow Cake", size: .small)
+        
+        totalSales += cone.size.rawValue
+        
+        if let topping = cone.topping {
+            print("Your \(cone.flavor) with \(topping) is \(cone.size.rawValue)")
+        } else {
+            print("Your \(cone.flavor) is \(cone.size.rawValue)")
+        }
+        
+        return cone
+    }
+}
```

Comment:
Here as well you would want to make it dynamic as this function will always create a Cone object with vanilla flavor, yellow cake toppings, and be small size. If you use flavor, topping, and size as parameters in your function and then use that to initialize a new cone object that will all you to pass in different values as needed when calling that function.